sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/AuthClient.java

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.jca;
+
+import com.azure.security.keyvault.jca.rest.OAuthToken;
+
+import java.util.HashMap;
+import java.util.logging.Logger;
+
+import static java.util.logging.Level.FINER;
+
+/**
+ * The REST client specific to getting an access token for Azure REST APIs.
+ */
+class AuthClient extends DelegateRestClient {
```

Comment:
MSAL would only be usable in this particular case if a) it is self-contained as a JAR and b) it can be properly shaded and relocated so customers are not impacted by versioning. As far as I can tell that is not the case hence it using its own client.